core/templates/components/forms/custom-forms-directives/image-uploader.component.ts

```diff
@@ -0,0 +1,151 @@
+// Copyright 2024 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Component for uploading images.
+ */
+
+import {EventEmitter, OnInit, Output} from '@angular/core';
+import {Component, Input} from '@angular/core';
+import {downgradeComponent} from '@angular/upgrade/static';
+import {NgbModal} from '@ng-bootstrap/ng-bootstrap';
+import {UrlInterpolationService} from 'domain/utilities/url-interpolation.service';
+import {ContextService} from 'services/context.service';
+import {ImageUploadHelperService} from 'services/image-upload-helper.service';
+import {ImageUploaderModalComponent} from './image-uploader-modal.component';
+
+export interface ImageUploaderParameters {
+  disabled: boolean;
+  maxImageSizeInKB: number;
+  imageName: string;
+  orientation: string;
+  bgColor: string;
+  allowedBgColors: string[];
+  allowedImageFormats: string[];
+  aspectRatio: string;
+  filename?: string;
+  previewTitle?: string;
+  previewDescription?: string;
+  previewDescriptionBgColor?: string;
+  previewFooter?: string;
+  previewImageUrl?: string;
+}
+
+@Component({
+  selector: 'oppia-image-uploader',
+  templateUrl: './image-uploader.component.html',
+})
+export class ImageUploaderComponent implements OnInit {
+  @Output() updateBgColor: EventEmitter<string> = new EventEmitter();
+  @Output() updateFilename: EventEmitter<string> = new EventEmitter();
+  @Output() imageSave: EventEmitter<Blob> = new EventEmitter();
+
+  @Input() imageUploaderParameters!: ImageUploaderParameters;
+
+  dimensions!: {height: number; width: number};
```

Comment:
This should be optional too, no? When there's no image there are no dimensions.

{0, 0} as an absent value is not a good approach. It's less explicit and also is not expressed in types

Reply:
this is kind of redundant we already can get dimension from the modal and generate the filename. so removed it.